### PR TITLE
Refactor merge fix

### PR DESCRIPTION
**Pull request by Jack-kelly-22**
refactor and cleanup warnings

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-    "workbench.colorTheme": "Noctis Azureus"
+    "workbench.colorTheme": "Noctis Obscuro"
```
Comment (akwong189) [inline .vscode/settings.json]: technically, we don't need this folder because it is someone's local settings

File: src/ArticleCard.js [hunk 2]
```diff
@@ -1,39 +1,104 @@
...
-    props.likeArticle(article.id);
+    like_article(article_id);
+  }
+
+  function user_viewed() {
+    let token = sessionStorage.getItem("access_token");
+    if (token.length !== 1) {
+      let data = { action: "add", article_id: article.id };
+      let head = { headers: { Authorization: "Bearer " + token } };
+      axios.post("http://127.0.0.1:5000/view", data, head);
```
Comment (akwong189) [inline src/ArticleCard.js]: does this url need to access to the backend?

File: src/ArticleCard.js [hunk 3]
```diff
@@ -1,39 +1,104 @@
...
+
+  function post_comment() {
+    let data = { action: "add", comment: new_comment, article_id: article.id };
+    let head = {
+      headers: {
+        Authorization: "Bearer " + sessionStorage.getItem("access_token"),
+      },
+    };
+    console.log(head);
+    axios.post("http://127.0.0.1:5000/comment", data, head).then((result) => {
```
Comment (akwong189) [inline src/ArticleCard.js]: same here

File: src/CommentCol.js [hunk 4]
```diff
@@ -90,7 +90,7 @@ export function CommentCol(props){
 CommentCol.defaultProps = {
```
Comment (akwong189) [inline src/CommentCol.js:90]: static comments?

File: src/EditProfile.js [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+    let data = {
+      bio: bio,
+      first_name: first_name,
+      last_name: last_name,
+      profile_pic: profile_pic,
+      password: password,
+      email: email,
+    };
+    axios
+      .post("http://127.0.0.1:5000/edit_profile", data, head)
```
Comment (akwong189) [inline src/EditProfile.js]: using a local url, would this affect anything during deployment?

File: src/FollowUserList.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+    }
+
+    function follow_user(user_id){
+        let token = sessionStorage.getItem('access_token');
+        let data = {user_id:user_id,action:'follow'};
+        let head = {headers:{Authorization:"Bearer "+ token}}
+
+    const [users, setUsers] = useState([]);    
+    axios.post("http://127.0.0.1:5000/follow_user",data,head);
```
Comment (akwong189) [inline src/FollowUserList.js]: local url

File: src/Login.js [hunk 7]
```diff
@@ -1,126 +1,130 @@
...
+    sessionStorage.clear();
+  }
 
-    return (
-        <div className="outer">
-            <div className='inner'>
-                <form>
+  function signUp() {
+    axios
+      .post("http://127.0.0.1:5000/newUser", {
```
Comment (akwong189) [inline src/Login.js]: local url

File: src/Login.js [hunk 8]
```diff
@@ -1,126 +1,130 @@
...
+  }
 
-                    <div className="form-group">
-                        <label>Email</label>
-                        <input onChange = {v=>setEmail(v.target.value)}type="email" className="form-control" placeholder="Enter email" />
-                    </div>
+  function login() {
+    console.log("starting login...");
+    axios
+      .post("http://127.0.0.1:5000/login", { email: email, password: password })
```
Comment (akwong189) [inline src/Login.js]: local url

File: src/Profile.js [hunk 9]
```diff
@@ -1,76 +1,130 @@
...
+    const [key, setKey] = useState('home');
+    // const [articles, setArticles] = useState([]);
+    const [articleCards,setArticleCards] = useState(null);
+
+    
+    useEffect(()=> {
+        let token = sessionStorage.getItem('access_token');
+        // let data = {user_id:user_id,action:'follow'};
+        let head = {headers:{Authorization:"Bearer "+ token}}
+        axios.post('http://127.0.0.1:5000/articles',{article_ids:user.reposted_articles},head).then( result => {
```
Comment (akwong189) [inline src/Profile.js]: local url

File: src/SearchBar.js [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+  const [user_ids, set_user_ids] = useState([]); 
+
+
+  function search_user(search_string) {
+    let token = sessionStorage.getItem("access_token");
+    let head = { headers: { Authorization: "Bearer " + token } };
+    let data = {username: search_string};
+    axios
+      .post(
+        "http://127.0.0.1:5000/search_user",
```
Comment (akwong189) [inline src/SearchBar.js]: local url

File: src/Social.js [hunk 11]
```diff
@@ -0,0 +1,50 @@
+import axios from "axios";
+
+export async function follow(user_id) {
+  let token = sessionStorage.getItem("access_token");
+  let data = { user_id: user_id, action: "follow" };
+  let head = { headers: { Authorization: "Bearer " + token } };
+  console.log("User follow attempt");
+  axios.post("http://127.0.0.1:5000/follow_user", data, head);
```
Comment (akwong189) [inline src/Social.js]: local url

File: src/Social.js [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+export async function get_users(user_ids) {
+  let res = null;
+  let token = sessionStorage.getItem("access_token");
+  let data = { user_ids: user_ids, action: "follow" };
+  let head = { headers: { Authorization: "Bearer " + token } };
+  axios.post("http://127.0.0.1:5000/users", data, head).then((result) => {
```
Comment (akwong189) [inline src/Social.js]: local url

File: src/Social.js [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+    }
+  });
+  console.log("recommended: ", res);
+}
+
+export async function repost_article(articleId) {
+  let token = sessionStorage.getItem("access_token");
+  let data = { action: "add", article_id: articleId };
+  let head = { headers: { Authorization: "Bearer " + token } };
+  axios.post("http://127.0.0.1:5000/repost", data, head);
```
Comment (akwong189) [inline src/Social.js]: local url

File: src/Social.js [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+export function save_article(articleId, saved) {
+  let data = { action: "add", article_id: articleId };
+  let token = sessionStorage.getItem("access_token");
+  let head = { headers: { Authorization: "Bearer " + token } };
+  if (saved) {
+    data.action = "delete";
+  }
+  axios.post("http://127.0.0.1:5000/save", data, head);
```
Comment (akwong189) [inline src/Social.js]: local url

File: src/Social.js [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+  let head = { headers: { Authorization: "Bearer " + token } };
+  axios.post("http://127.0.0.1:5000/repost", data, head);
+}
+
+export function like_article(articleId) {
+  let token = sessionStorage.getItem("access_token");
+  let data = { action: "add", article_id: articleId,};
+  console.log("data:", data);
+  let head = { headers: { Authorization: "Bearer " + token } };
+  axios.post("http://127.0.0.1:5000/like", data, head);
```
Comment (akwong189) [inline src/Social.js]: using local url

File: src/ThreadPage.js [hunk 16]
```diff
@@ -1,39 +1,49 @@
...
+    let topic =
+      window.location.href.substring(window.location.href.lastIndexOf("/") + 1)
+        .length === 0
+        ? ""
+        : window.location.href.substring(
+            window.location.href.lastIndexOf("/") + 1
+          );
+    console.log(window.location.href);
+    let data = { topic: topic, n: 50 };
+    axios.post("http://127.0.0.1:5000/threads", data, head).then((result) => {
```
Comment (akwong189) [inline src/ThreadPage.js]: using local url

**Review by akwong189 — CHANGES_REQUESTED** [review]
there are local url references in a few files that may cause issues when deployed to heroku

**Comment by Jack-kelly-22** [comment]
updated urls

**Review by akwong189 — APPROVED** [review]
these 2 files look good; however, other files still need to be changed

**Review by akwong189 — CHANGES_REQUESTED** [review]
mistakenly approved, still a few files with local urls

File: src/Social.js (hunk 12 above)
Comment (akwong189) [reply src/Social.js]: was fixed

**Review by akwong189 — APPROVED** [review]
All urls have been resolved

**Pull request merged**